ledger/sandbox/src/main/scala/com/digitalasset/platform/sandboxnext/Runner.scala

```diff
@@ -120,43 +121,46 @@ class Runner(config: SandboxConfig) extends ResourceOwner[Port] {
                   config.metricsReporter,
                   config.metricsReportingInterval,
                 )
-                _ <- startupMode match {
-                  case StartupMode.MigrateAndStart =>
-                    ResourceOwner.successful(())
-                  case StartupMode.ResetAndStart =>
-                    // Resetting through Flyway removes all tables in the database schema.
-                    // Therefore we don't need to "reset" the KV Ledger and Index separately.
-                    ResourceOwner.forFuture(() => new FlywayMigrations(indexJdbcUrl).reset())
-                }
                 timeServiceBackend = timeProviderType match {
                   case TimeProviderType.Static =>
                     Some(TimeServiceBackend.simple(Instant.EPOCH))
                   case TimeProviderType.WallClock =>
                     None
                 }
+                isReset = startupMode == StartupMode.ResetAndStart
                 readerWriter <- new SqlLedgerReaderWriter.Owner(
                   initialLedgerId = specifiedLedgerId,
                   participantId = ParticipantId,
                   metricRegistry = metrics,
                   jdbcUrl = ledgerJdbcUrl,
+                  resetOnStartup = isReset,
                   timeProvider = timeServiceBackend.getOrElse(TimeProvider.UTC),
                   seedService = SeedService(seeding),
                   stateValueCache = caching.Cache.from(
                     caching.Configuration(maximumWeight = MaximumStateValueCacheSize)),
+                  engine = engine
                 )
                 ledger = new KeyValueParticipantState(readerWriter, readerWriter, metrics)
                 readService = new TimedReadService(ledger, metrics, ReadServicePrefix)
                 writeService = new TimedWriteService(ledger, metrics, WriteServicePrefix)
                 ledgerId <- ResourceOwner.forFuture(() =>
                   readService.getLedgerInitialConditions().runWith(Sink.head).map(_.ledgerId))
-                _ <- ResourceOwner.forFuture(() =>
-                  Future.sequence(config.damlPackages.map(uploadDar(_, writeService))))
+                _ <- if (isReset) {
+                  ResourceOwner.unit
+                } else {
+                  ResourceOwner
+                    .forFuture(() =>
+                      Future.sequence(config.damlPackages.map(uploadDar(_, writeService))))
+                    .map(_ => ())
+                }
```

Comment:
I think it'd be useful to add a test case to ensure that the packages are available and can be both listed and used after a reset.

Reply:
Added a test case in b4d5a7f.